cou/utils/juju_utils.py

```diff
@@ -509,3 +520,304 @@ def _check_time(self) -> None:
             raise TimeoutException(
                 f"Model {self.model_name} has not stabilized after {self.timeout} seconds."
             )
+
+
+def retry(
+    function: Optional[Callable] = None,
+    timeout: int = DEFAULT_TIMEOUT,
+    no_retry_exception: Optional[tuple] = None,
```

Comment:
and should we maybe rename it to a plural `no_retry_exceptions` since it accepts a tuple?